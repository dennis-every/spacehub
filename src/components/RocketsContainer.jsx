import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from './Rocket';

const RocketsContainer = () => {
  const { rockets } = useSelector((state) => state.rockets);

  console.log('Rockets inside container ', rockets);
  return (
    <div>
      {rockets.map((item) => (
        <Rocket
          key={item.rocketId}
          rocketId={item.rocketId}
          image={item.image}
          name={item.name}
          description={item.description}
          reserved={item.reserved}
        />
      ))}

    </div>
  );
};

export default RocketsContainer;
