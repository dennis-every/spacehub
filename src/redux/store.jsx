import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketsReducer,
  },
});

export default store;
