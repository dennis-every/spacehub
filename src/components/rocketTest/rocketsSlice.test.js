import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import RocketsContainer from '../RocketsContainer';
import store from '../../redux/store';

const mockStore = configureStore([]);

describe('Rockets redux state tests', () => {
  it('Expect to initially set rockets to an empty object', () => {
    const state = store.getState().rockets;
    const initialState = {
      rockets: [], isLoading: true,
    };
    expect(state).toEqual(initialState);
  });
  it('Expect to display one rocket from mockStore', () => {
    const rockets = [
      {
        rocketId: 'falcon1',
        name: 'Falcon 1',
        description:
          'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.',
        reserved: true,
      },
    ];
    const state = {
      rockets: {
        rockets,
        isLoading: false,
      },
    };
    const store = mockStore(state);
    render(
      <Provider store={store}>
        <RocketsContainer />
      </Provider>,
    );
    const name = screen.getByText('Falcon 1');
    expect(name).toBeInTheDocument();
    expect(store.getState).toMatchSnapshot();
  });
});
