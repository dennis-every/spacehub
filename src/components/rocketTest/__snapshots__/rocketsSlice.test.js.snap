// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets redux state tests Expect to display one rocket from mockStore 1`] = `[Function]`;
